### DISCUSSION TITLE
Google tag conversion code is working fine in generated URL but not with domain mapped url

### DISCUSSION THREAD

**Discussion by Ajay-ML** [vercel/community#1315] [Help]
Hi,

In one of our Next JS project we added Google Tag conversion code via script tag and deployed it to production. Later we realized that there was a bug where conversion call was happening in each and every page instead of happening in a particular flow. So we fixed the issue and deployed the bug fixes to production again.

Now the issue is, if we access the app via Generated URL (dynamic url which Vercel will generate for each deployment) tag conversions are not happening on every page load and it is working as expected. If we test the app with custom domain mapped URL I can still see the old behaviour where tag conversion is happening on each page load i.e. the latest fixes are not reflecting if we access via domain, it is reflecting only if we access via generated URL.

Please can some one suggest what might be issue or how to resolve this?

Tried below things already, still no luck.
1. Tried redeploying without selecting cache
2. Did some UI changes in code and did prod deployment
3. Cleared cache and tried
4. Tried in incognito window

Regards,
Ajay

**Comment by amyegan** [comment]
Hi @Ajay-ML! Can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or explain your current strategy in more detail? There are a lot of things that could be going wrong, and knowing more about your project setup will help narrow down the possibilities

**Reply by Ajay-ML** [reply]
Hi @amyegan ,

Below are the details,

1. We embedded below Google tag conversion code in app.tsx file of our next js application.
      `gtag('event', 'conversion', {
                 'send_to': 'Google_ADS_ID}/Sign-up_Conversion_Label',
              });`

2. As app.tsx is the main component in our app which is rendered to the DOM by ReactDOM. Hence this tag conversion code was getting triggered for each page refresh. This was not the requirement, it was suppose to call conversion code in only particular scenario i.e. Signup
3. We fixed this bug not to call conversion code on page refresh so we removed the code from app.tsx and moved the conversion code into Signup flow code and deployed code to production
4. When we access application via Vercel unique URL generated for the deployment I can see the code changes is working fine and the tag conversion call is happening only in Signup flow. But unfortunately when I access the app from the custom domain mapped, it still loads old code (tag conversion happening on each page refresh) and not reflecting the new changes deployed.
5. So Google tag conversion code is working with Vercel generated URL but not with domain
6. We have added below gtag code in app.tsx so that it is available in all pages
  `<script
  async
  window.dataLayer = window.dataLayer || [];
                window.gtag = function gtag(){window.dataLayer.push(arguments);}
                gtag('js', new Date());
                gtag('config', '${Google_ADS_ID}');
              `
            }}
          />`

Let me know if these information is sufficient or do you need any further info.


Tried below things already, still no luck.

Tried redeploying without selecting cache
Did some UI changes in code and did prod deployment
Cleared cache and tried
Tried in incognito window

Regards,
Ajay

**Reply by amyegan** [reply]
Thanks for sharing this info. Do you have a [minimal reproduction](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or example deployment links you can share so I can see the problem in action?
